include/text.php

```diff
@@ -718,6 +714,10 @@ function logger($msg, $level = 0) {
 		return;
 	}
 
+	$debugging = get_config('system','debugging');
+	$logfile   = get_config('system','logfile');
+	$loglevel = intval(get_config('system','loglevel'));
```

Comment:
Watch out, you are defining variables that are used in the previous condition. The condition needs to be broken back into two conditions blocks, one with `$a->module` and `$db->connected` and the other one with the settings conditions.